fix(insights): Improve load more for breakdowns

webjunkie opened the pull request:
## Problem

It was only working if you selected "display other", because presumably you otherwise couldn't tell if more breakdown values are available.

Fixes #23554

## Changes

- make it show up in all cases
- adjust the wording to have the actual values - since you also modify the query with this and it's persisted when you save
- use separate `hasMore` flag that is derived from "other" breakdowns being available. For this to work we ignore the `hide_other` flag and always calculate them, but hide the label+value later.

<img width="758" alt="image" src="https://github.com/user-attachments/assets/12afa640-b1ce-43b4-b22e-b3157ab1035a">


## How did you test this code?

Tested locally

webjunkie committed:
Improve load more for breakdowns

It was only working if you selected "display other"
presumably this was because of the filter conversion?

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

thmsobrmlr commented on frontend/__snapshots__/scenes-app-experiments--running-trend-experiment--dark.png:
Seems this affects experiment results as well, where we don't need it.

thmsobrmlr commented on frontend/__snapshots__/scenes-app-insights--trends-area-breakdown--dark--webkit.png:
In general we should probably only show this if there are more breakdown values.

webjunkie committed:
Merge branch 'master' into fix/breakdown-limit-load-more

# Conflicts:
#	frontend/__snapshots__/scenes-app-insights--trends-area-breakdown-edit--dark--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-area-breakdown-edit--dark.png
#	frontend/__snapshots__/scenes-app-insights--trends-area-breakdown-edit--light--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-area-breakdown-edit--light.png
#	frontend/__snapshots__/scenes-app-insights--trends-bar-breakdown-edit--dark--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-bar-breakdown-edit--dark.png
#	frontend/__snapshots__/scenes-app-insights--trends-bar-breakdown-edit--light--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-bar-breakdown-edit--light.png
#	frontend/__snapshots__/scenes-app-insights--trends-line-breakdown-edit--dark--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-line-breakdown-edit--dark.png
#	frontend/__snapshots__/scenes-app-insights--trends-line-breakdown-edit--light--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-line-breakdown-edit--light.png
#	frontend/__snapshots__/scenes-app-insights--trends-pie-breakdown-edit--dark--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-pie-breakdown-edit--dark.png
#	frontend/__snapshots__/scenes-app-insights--trends-pie-breakdown-edit--light--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-pie-breakdown-edit--light.png
#	frontend/__snapshots__/scenes-app-insights--trends-table-breakdown-edit--dark--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-table-breakdown-edit--dark.png
#	frontend/__snapshots__/scenes-app-insights--trends-table-breakdown-edit--light--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-table-breakdown-edit--light.png
#	frontend/__snapshots__/scenes-app-insights--trends-value-breakdown-edit--dark--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-value-breakdown-edit--dark.png
#	frontend/__snapshots__/scenes-app-insights--trends-value-breakdown-edit--light--webkit.png
#	frontend/__snapshots__/scenes-app-insights--trends-value-breakdown-edit--light.png

webjunkie committed:
Add hasMore to indicate if breakdown values have been truncated

Derive from breakdown other option, which we always query for and then
just discard later on.

webjunkie commented:
@thmsobrmlr I've added a relatively quick approach/workaround to tell if there are actually more values. Can you check again? Also for experiments, I checked one and now the load more isn't there. But maybe that depends on certain conditions, so would be great if you can also check that.

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

webjunkie committed:
Merge branch 'master' into fix/breakdown-limit-load-more

thmsobrmlr committed:
add comment

thmsobrmlr committed:
gracefully handle missing breakdownFilter

thmsobrmlr reviewed (changes requested):
Thanks for the follow-up @webjunkie! Works great now.

Did a couple of stylistic changes, hope you don't mind.

There's only one thing blocking this from merging now: The button only appears on a new insight, but not after saving. I'm assuming this has something to do with the insight serializer not returning the `hasMore` or similar.

thmsobrmlr commented:
> Also for experiments, I checked one and now the load more isn't there. But maybe that depends on certain conditions, so would be great if you can also check that.

This was just an artifact of the previous implementation where we would always display the button regardless if there are more values. Experiments only have two values and as such it's working there now.

webjunkie committed:
Merge branch 'master' into fix/breakdown-limit-load-more

webjunkie committed:
Fix has more

webjunkie committed:
Fix lemon table all series check

Fixes #24307

webjunkie commented:
> with the insight serializer not returning the hasMore or similar.

Okay fixed this now. The `hasMore` is next to the results... which means it needs to go into the serializer as well. This whole area is a bit cumbersome. I wonder why no other insights had specific fields that needed to be included. We have `hasMore` also for other query responses, but it seems it's mostly actors responses that are not going to be the result of a saved insight 🤔

thmsobrmlr commented:
> > with the insight serializer not returning the hasMore or similar.
> 
> Okay fixed this now. The `hasMore` is next to the results... which means it needs to go into the serializer as well. This whole area is a bit cumbersome. I wonder why no other insights had specific fields that needed to be included. We have `hasMore` also for other query responses, but it seems it's mostly actors responses that are not going to be the result of a saved insight 🤔

Yeah it's quite annoying. We did have other insights with problems here and probably still have some edge cases that are unhandled..

github-actions[bot] committed:
Update query snapshots

webjunkie committed:
Merge branch 'master' into fix/breakdown-limit-load-more

webjunkie committed:
Fix test
